examples/hyperparameter_tuning/dynamic_pipelines/dynamic_pipeline.py

```diff
@@ -0,0 +1,51 @@
+from typing import Any, Type, TypeVar
+
+from zenml.pipelines import BasePipeline
+from zenml.steps import BaseStep
+
+DP = TypeVar("DP", bound="DynamicPipeline")
+
+
+class DynamicPipeline(BasePipeline):
```

Comment:
If I am not missing any crucial details, I have a question about the effect of this class as well. I believe, previously, this class had other methods which played a critical role, however, in its current state, it is quite straightforward and the main functionality lies only in the `as_template_of` method. 

If we take into account what happens within the `__init__` and `connect` methods of the `HyperparameterTuning`, we can think about whether we can generalize the creation of these dynamic steps. If this is possible, we can merge that functionality with the `as_template_of` method and even create another `@dynamic_pipeline` decorator in the future. What do you think?

Reply:
Other than the `as_template_of` method, the purpose of this class is also the initialization of STEP_SPEC. I considered just creating simple methods and using them in a pipeline that inherits `BasePipeline`, but I decided that this solution encapsulates the dynamic pipeline functionality more elegantly.
I agree that for adding dynamic pipelines in ZenML, we need a slightly different solution so the connect method can be defined with a dynamic pipeline decorator, and the dynamic steps definition would be injected into it some other way (maybe a `DynamicSteps` class).

Reply:
@bcdurak did you get your answer? Can I mark this as resolved?

Reply:
@nitay93 thank you for the explanation. Yes, it is much clearer now and you can mark it as resolved.